ipinfo/cmd_login.go

```diff
@@ -67,3 +80,20 @@ func cmdLogin() error {
 
 	return nil
 }
+
+func checkToken(tok string) (bool, error) {
+	// Make a request to the /me ep
+	res, err := http.Get("https://ipinfo.io/me?token=" + tok)
+	if err != nil {
+		return false, err
+	}
+
+	// Read the body of the response
+	b, err := io.ReadAll(res.Body)
+	if err != nil {
+		return false, err
+	}
+
+	// The response should not contain the "error" field.
+	return !strings.Contains(string(b), "error"), nil
```

Comment:
We definitely need something more robust than just searching for "error" anywhere in the raw JSON string. If you do a proper JSON decode into a generic map, you can then ensure the `error` key is not there.

```go
m := make(map[string]interface{})
err = json.Unmarshal(b, m)
if err != nil {
	return false, err
}

_, hasErr := m["error"]
return !hasErr, nil
```

Reply:
(Another benefit of doing it this way is we can in the future print out the user's limits/etc from this map when the token is valid).